tools/infrasys/src/keys.rs

```diff
@@ -0,0 +1,130 @@
+/// Output: Edits KMSConfig fields in place after creating new keys
+pub async fn create_keys(signing_key_config: &mut SigningKeyConfig) -> Result<()> {
+    // An extra check even through these parameters are checked earlier in main.rs
+    check_signing_key_config(signing_key_config)?;
+    match signing_key_config {
+        SigningKeyConfig::file { .. } => (),
+        SigningKeyConfig::kms { config, .. } => {
+            config
+                .as_mut()
+                .context(error::MissingConfig {
+                    missing: "config field for a kms key",
+                })?
+                .create_kms_keys()
+                .await?;
+        }
+        SigningKeyConfig::ssm { .. } => (),
+    }
+    Ok(())
+}
+
+pub fn check_signing_key_config(signing_key_config: &SigningKeyConfig) -> Result<()> {
+    match signing_key_config {
+        SigningKeyConfig::file { .. } => (),
+        SigningKeyConfig::kms { config, .. } => {
+            let unwrapped_config = config.as_ref().context(error::MissingConfig {
+                missing: "config field for a kms key",
+            })?;
+
+            match (
+                unwrapped_config.available_keys.is_empty(),
+                unwrapped_config.regions.is_empty(),
+                unwrapped_config.key_alias.as_ref(),
+            ) {
+                (true, true, None) => error::KeyConfig {
+                    missing: "an available_key or region/key_alias",
+                }
+                .fail()?,
+                (_, false, None) => error::KeyConfig {
+                    missing: "key_alias",
+                }
+                .fail()?,
+                (_, true, Some(..)) => error::KeyConfig { missing: "region" }.fail()?,
+                _ => (),
+            };
+        }
+        SigningKeyConfig::ssm { .. } => (),
+    }
+    Ok(())
+}
+
+/// Must create a trait because can't directly implement a method for an struct in an external crate like KMSKeyConfig (which lives in pubsys-config/lib.rs)
+#[async_trait]
+trait KMSKeyConfigExt {
+    async fn create_kms_keys(&mut self) -> Result<()>;
+}
+
+/// Creates new KMS keys using cloudformation in regions specified
+/// Input Conditions: Alias+Region or AvailableKeys must be specified
+/// Output: Populates KMSKeyConfig with information about resources created
+/// available-keys starts as a map of pre-existing keyids:regions and will end as a map of pre-existing and generated keyids:regions
+/// key-stack-arns starts empty and will end as a map of keyids:stackarn if new keys are created
+#[async_trait]
+impl KMSKeyConfigExt for KMSKeyConfig {
+    async fn create_kms_keys(&mut self) -> Result<()> {
+        // Generating new keys (if regions is non-empty)
+        for region in self.regions.iter() {
+            // Unwrapping key_alias here, because if regions is defined, key_alias must be
+            let stack_name = format!("TUF-KMS-{}", self.key_alias.as_ref().unwrap());
```

Comment:
I know technically we would never panic due to the `unwrap` here since we're checking earlier, but we should still avoid unwrapping since we can just easily substitute this with an `error::KeyConfig`.
```suggestion
            let key_alias =  self.key_alias.as_ref().context(error::KeyConfig { ... })?;
            let stack_name = format!("TUF-KMS-{}", key_alias);
```

Also there's an unwrap a few lines below as well.